src/mem3_shards.erl

```diff
@@ -360,3 +397,15 @@ cache_clear(St) ->
     true = ets:delete_all_objects(?SHARDS),
     true = ets:delete_all_objects(?ATIMES),
     St#st{cur_size=0}.
+
+filter_shards_by_name(Name, Shards) ->
+    filter_shards_by_name(Name, [], Shards).
+
+filter_shards_by_name(_, Shards, []) ->
+    Shards;
+filter_shards_by_name(Name, Shards, [#ordered_shard{name=Name}=S|Ss]) ->
+    filter_shards_by_name(Name, [S|Shards], Ss);
+filter_shards_by_name(Name, Shards, [#shard{name=Name}=S|Ss]) ->
+    filter_shards_by_name(Name, [S|Shards], Ss);
+filter_shards_by_name(Name, Shards, [_|Ss]) ->
+    filter_shards_by_name(Name, Shards, Ss).
```

Comment:
I found it confusing to pass `Shards` in as the third argument in the 2-arity implementation and then bind `Shards to the second argument internally. Maybe change to `Matches` or something there?